python/objToJSON.c

```diff
@@ -579,7 +592,7 @@ void Object_beginTypeContext (JSOBJ _obj, JSONTypeContext *tc)
     return;
   }
   else
-  if (PyString_Check(obj))
+  if (PyString_Check(obj) && !PyObject_HasAttrString(obj, "__json__"))
```

Comment:
Hi @mitar, What's the use-case that warranted this particular change? It looks like this is a performance regression.